src/jit/lsraxarch.cpp

```diff
@@ -2736,9 +2801,8 @@ void LinearScan::TreeNodeInfoInitCmp(GenTreePtr tree)
 
     if (op1->TypeGet() != TYP_VOID)
     {
-        info->srcCount += GetOperandSourceCount(op1);
+        info->srcCount = appendBinaryLocationInfoToList(tree->AsOp());
     }
-    info->srcCount += GetOperandSourceCount(op2);
```

Comment:
Looks like the `srcCount` from the second op is missed here if the first op has void type.

Reply:
Weird; I can't imagine why op1 would ever be `TYP_VOID` (and I ran diffs after this set of changes). I'll look into that - thanks!

Reply:
I think that condition check was a mistake in the old code. I made the call to `appendBinaryLocationInfoToList()` unconditional, and there are zero diffs on both xarch targets. (Though I did have new diffs I'd introduced with changes to the ShiftRotate and ModDiv cases, which I've now fixed.)